src/emulator/functionsEmulatorRuntime.ts

```diff
@@ -638,11 +678,31 @@ async function RunHTTPS(
   });
 }
 
-function isFeatureEnabled(
-  frb: FunctionsRuntimeBundle,
-  feature: keyof FunctionsRuntimeFeatures
-): boolean {
-  return frb.disabled_features ? !frb.disabled_features[feature] : true;
+/*
+  This method attempts to help a developer whose code can't be loaded by suggesting
+  possible fixes based on the files in their functions directory.
+ */
+async function moduleResolutionDetective(frb: FunctionsRuntimeBundle): Promise<void> {
+  /*
+  These files could all potentially exist, if they don't then the value in the map will be
+  falsey, so we just catch to keep from throwing.
+   */
+  const clues = {
+    tsconfigJSON: await requireAsync("./tsconfig.json", { paths: [frb.cwd] }).catch(NoOp),
```

Comment:
why `await requireAsync` instead of just `require`?

Reply:
So in this case (and with the newly added `requireResolveAsync`) the value of being `async` is in cleanliness of control flow - it is not required because these operations are not async, but it helps when dealing with synchronous methods which may throw (like `require`). By wrapping these native methods in `async` functions then catching with a `NoOp`, we gain the ability to write one-liners which catch thrown errors and just end up `false`-y 

So this snippet, if it didn't use the Promise-ified version would be this...

```js
const clues: {[s: key]: any} = {};

try {
  clues["tsconfigJSON"] = require("./tsconfig.json", { paths: [frb.cwd] });
} catch (err) {
  // Optional explicit set to false since default value of undefined is false-y
  clues["tsconfigJSON"] = false;
};

try {
  clues["packageJSON"] = require("./package.json", { paths: [frb.cwd] });
} catch (err) {
  clues["packageJSON"] = false;
};
```

As opposed to the `async`d version

```js
const clues = {
    tsconfigJSON: await requireAsync("./tsconfig.json", { paths: [frb.cwd] }).catch(NoOp),
    packageJSON: await requireAsync("./package.json", { paths: [frb.cwd] }).catch(NoOp),
}
```

In either case, the clue value will be either an `any` from the `require` call or `false`.

We've shifted to this flow all over the place because we're constantly doing requires and resolutions which may fail but aren't really fatal.